### ISSUE TITLE
Concurrency exceptions at serializable isolation level with postgres

### ISSUE THREAD

**Issue by toolkit** [kagkarlsson/db-scheduler#295]
Hi there. Thanks for this great library, which has been really useful for our teams.

I have a question about concurrency exceptions seen with postgres (Amazon RDS Aurora)

Due to an architectural constraint, we have to set our default isolation level to serializable.

When we did this initially, we saw a lot of exceptions of the form

`PSQLException: ERROR: could not serialize access due to read/write dependencies among transactions `

We reasoned this could be reduced by using the lock-and-fetch polling strategy.

In many cases we no longer see this exception, but are seeing now exceptions of the form

`PSQLException: ERROR: could not serialize access due to concurrent update`

Could I ask:

- Would these be db scheduler threads competing to run the same task, so that we could consider these exceptions benign?
- If not, is there a safe isolation level for updating scheduled_tasks?
- If so, what is the preferred way to specify a lower isolation level for DB Scheduler?

Once again, thanks for all your great work 👍🏻

**Comment by kagkarlsson** [comment]
> Is this benign? That is, (distributed) threads competing for the same task?

Yeah, that should be benign and caused by your stricter isolation level if I understand things correctly. See for example 
https://pganalyze.com/docs/log-insights/app-errors/U138

It looks like it does not reach db-schedulers built-in mechanism for guaranteeing that a single instance gets the lock.

What thread are you getting this exception in?

**Comment by toolkit** [comment]
Out of curiosity, would it be possible to define a property to override the isolation level that DB Scheduler uses (ie: not use the isolation level on the incoming datasource) ?

**Comment by kagkarlsson** [comment]
I had assumed it was caused by concurrent updates, but that should really not happen that frequently in detect-dead or heartbeat-threads. Is the exception frequent? Are you using very low interval for heartbeat or something like that? 
LArge number of executions, i.e. big table?
